contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,910 @@
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+        require(
+            _currentPrice >= sqrtRatioX96Tick0 &&
+                _currentPrice <= sqrtRatioX96Tick1,
+            "Not in expected tick range"
+        );
+
+        // If token addresses were reversed estimateAmount0 would
+        // be required here
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOETHAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(clPool),
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOETHAmount
+        );
+
+        uint256 _wethDiff = Math.max(poolWethShare, _wethSharePct) -
+            Math.min(poolWethShare, _wethSharePct);
+
+        if (_wethDiff < poolWethShareVarianceAllowed) {
+            emit PoolRebalanced(_wethSharePct, poolWethShare);
```

Comment:
🟣 In our current implementation of rebalance, we check this twice, thus we will emit two events. I guess that's okay.

Reply:
Yeah in theory we don't need to do it after depositing the liquidity. Though it doesn't make complete sense to do it again at the end of a rebalance function I somehow feel paranoid removing it.